### PR TITLE
InputCurrency

### LINKED ISSUE #213: Currency Input
## Overview

A specialized input to currencies.

## Purpose

Automatic punctuating numbers inside the input component using context or overwriting by properties

## DoD

- [x] Design
- [x] Component implementation
- [ ] Documentation

## Examples

https://www.npmjs.com/package/react-currency-input

### PR DESCRIPTION
**Pull request by ericreis**
Add **InputCurrency** new component for monetary input values. It uses [react-number-format](https://www.npmjs.com/package/react-number-format) package to deal with the number formatting.

This closes #213.

### CODE REVIEW

File: react/components/InputCurrency/README.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+initialState = { value: undefined }
+;<div>
+  <div className="mb5">
+    <InputCurrency
+      label="Small"
+      size="small"
+      placeholder="Type a monetary value"
+      currencySymbol="R$"
+      decimalSeparator=","
+      thousandSeparator="."
```
Comment (klzns) [inline react/components/InputCurrency/README.md]: Acho que isso td ja deveria ser resolvido pelo componente, passando apenas o currencyCode.

File: react/components/InputCurrency/index.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+          ...event.target,
+          value: '',
+        },
+      })
+    this.props.onClear && this.props.onClear(event)
+  }
+
+  render() {
+    const { currencySymbol, decimalSeparator, thousandSeparator } = this.props
+    console.log(decimalSeparator, thousandSeparator)
```
Comment (klzns) [inline react/components/InputCurrency/index.js]: 🔫

File: react/components/InputCurrency/index.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    return (
+      <div>
+        <NumberFormat
+          {...this.props}
+          inputPrefix={currencySymbol}
```
Comment (klzns) [inline react/components/InputCurrency/index.js]: Euro é sufixo

File: react/components/InputCurrency/index.js [hunk 4]
```diff
@@ -65,17 +74,13 @@ InputCurrencyWithRef.propTypes = {
   size: PropTypes.string,
   defaultValue: PropTypes.number,
   value: PropTypes.number,
-  /** Currency symbol used as the input prefix. */
-  currencySymbol: PropTypes.string,
-  /** Character used for decimal separation. Eg: '.' in US$ 1,325.25. */
-  decimalSeparator: PropTypes.string,
-  /** Character used for thounsand separation. Eg: ',' in US$ 1,325.25. */
-  thousandSeparator: PropTypes.string,
+  locale: PropTypes.string,
```
Comment (klzns) [inline react/components/InputCurrency/index.js]: I would keep `locale` and `currencyCode` as `isRequired`.

Comment (klzns) [reply react/components/InputCurrency/index.js]: And would remove defaultProps

Comment (ericreis) [reply react/components/InputCurrency/index.js]: ✅

**Pull request merged**